### PR TITLE
More spec cleanup + bug fixes

### PR DESCRIPTION
**Pull request by mezis**
- [x] Addresses all Code Climate issues
- [x] Adds Code Climate configuration (with defaults)
- [x] Fixes `APIClient#fget` and `#with_response`; adds missing test coverage.
- [x] Fixes greedyness + recursion in collection responses

### CODE REVIEW

**Comment by nronas** [comment]
👍

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,213 @@
+ecmaFeatures:
```
Comment (danielcooper) [inline .eslintrc]: Do we have any javascript?

**Review by danielcooper — APPROVED** [review]

File: .eslintrc (hunk 1 above)
Comment (mezis) [reply .eslintrc]: Nope :)
I'll clean this up.

**Comment by mezis** [comment]
I rebased this and force pushed for trigger Travis, which was in a 💩 state after the big outage yesterday. Sorry.

**Pull request merged**